app/views/steps/_form.html.erb

```diff
@@ -59,11 +59,67 @@
 </div>
 
 <div class="form-group">
-  <div>
-    <p>
-      <%= step.broken_links %>
-    </p>
-  </div>
+  <% if step.broken_links && step.broken_links.count > 0 %>
+    <div class="broken-links">
+      <div class="panel-group panel-warning" id="accordion" role="tablist" aria-multiselectable="true">
+        <div class="panel-heading">
+          <%= render 'shared/steps/broken-links-summary', step: step %>
+        </div>
+        <% step.broken_links.each.with_index do |link, index| %>
+          <div class="panel panel-warning">
+            <div class="panel-heading" role="tab" id="brokenLinkHeading<%= index %>">
+              <h4 class="panel-title">
+                <a role="button" data-toggle="collapse" data-parent="#accordion" href="#brokenLink<%= index %>" aria-expanded="true" aria-controls="brokenLink<%= index %>">
+                  <%= link['uri'] %>
+                </a>
+              </h4>
+            </div>
+            <div id="brokenLink<%= index %>" class="panel-collapse collapse" role="tabpanel" aria-labelledby="brokenLinkHeading<%= index %>">
+              <div class="panel-body">
+                <dl class="dl-horizontal">
+                  <dt>URI</dt>
+                  <dd><%= link['uri'] %></dd>
+                  
+                  <dt>Status</dt>
+                  <dd><%= link['status'] %></dd>
+                  
+                  <dt>Last checked</dt>
+                  <dd><%= link['checked'].to_datetime.to_formatted_s(:rfc822) %></dd>
+                  
+                  <dt>Problem summary</dt>
+                  <dd><%= link['problem_summary'] %></dd>
+                  
+                  <% if link['errors'] && link['errors'].count > 0 %>
+                    <dt>Errors</dt>
+                    <dd>
+                      <ul class="broken-links--list">
+                        <% link['errors'].each do |error| %>
+                          <li><%= error %></li>
```

Comment:
Can we confirm whether there'll ever be more than one error for a broken link (@jonodrew?). It looks odd styled as a list with a single list item. Even if we could get more than one error, if for the majority of links there'll only be one, could this just be output as paragraphs or each followed by a `br`?

<img width="319" alt="screen shot 2018-08-10 at 08 36 14" src="https://user-images.githubusercontent.com/861310/43944916-7ef59e70-9c78-11e8-91f9-28f566327382.png">

Reply:
I'll restyle it